### DISCUSSION TITLE
Shlink + Google Cloud Run (GCP)

### DISCUSSION THREAD

**Discussion by leonardo2204** [shlinkio/discussions#71] [Help wanted] [answered]
#### How Shlink is set-up

* Shlink Version: 3.0.3
* PHP Version: 8.1.3
* How do you serve Shlink: Docker image in Cloud Run (GCP)
* Database engine used: PostgreSQL

#### Summary

Hey, first of all, thanks for this great project! My question is, since I`m running shlink inside cloud run, the background tasks may get killed and re-created without user control. What are the downsides? Am I missing something?
And should I be using redis since it`s running on multi cloud-runs?

Thanks!

**Comment by acelaya** [comment, marked as answer]
The things that Shlink could potentially do in the background are:

* Notify visits to a mercure hub instance, RabbitMQ instance or webhooks, for real-time updates
* Update the GeoLite2 DB if the existing instance is older than 35 days (a new version is released every month approximately)

The first is not relevant if you are not using any of those systems for real-time updates.

The last one is relatively important, but if the task fails, Shlink will continue locating visits with the older instance.

**Reply by leonardo2204** [reply]
Cool, thanks for the information! What about redis? Should I use it as I'm running multiple container in parallel?

Thanks!
